internal/services/reservations_service.go

```diff
@@ -20,6 +21,10 @@ var (
 
 // CreateReservation dispatches requests to type provider specific handlers
 func CreateReservation(w http.ResponseWriter, r *http.Request) {
+	if !config.LaunchEnabled(r.Context()) {
+		writeUnauthorized(w, r)
+	}
+
 	pType := models.ProviderTypeFromString(chi.URLParam(r, "TYPE"))
 	switch pType {
 	case models.ProviderTypeNoop:
```

Comment:
This is the place - when launching is disabled for an account or globally, then it returns unauthorized. Listing reservations is possible (through API) tho.